src/multivariate/product.jl

```diff
@@ -38,8 +38,13 @@ end
 
 _rand!(rng::AbstractRNG, d::Product, x::AbstractVector{<:Real}) =
     map!(Base.Fix1(rand, rng), x, d.v)
-_logpdf(d::Product, x::AbstractVector{<:Real}) =
-    sum(n->logpdf(d.v[n], x[n]), 1:length(d); init=0.0)
+function _logpdf(d::Product, x::AbstractVector{<:Real})
+    if isempty(dists)
```

Comment:
@devmotion are you sure that works? `dists` is only assigned below...

Reply:
It didn't, it was a mix of two versions I had played around with locally 😅 But I fixed it in a subsequent commit in this PR.